lib/modulesync/repository.rb

```diff
@@ -72,14 +72,16 @@ def clone
       @git = Git.clone(@remote, @directory)
     end
 
-    def prepare_workspace(branch)
+    def prepare_workspace(branch:, operate_offline:)
       if cloned?
         puts "Overriding any local changes to repository in '#{@directory}'"
-        git.fetch 'origin', prune: true
+        git.fetch 'origin', prune: true unless operate_offline
         git.reset_hard
         switch(branch: branch)
-        git.pull('origin', branch) if remote_branch_exists?(branch)
+        git.pull('origin', branch) if remote_branch_exists?(branch) && !operate_offline
```

Comment:
I'd swap the order and first check `!operate_offline`. I'm not sure if `remote_branch_exists?` reaches out to the network, but it's certainly cheaper to check a boolean first.

Reply:
Good catch, thanks. Suggestion applied.